R/pkg/inst/worker/daemon.R

```diff
@@ -31,7 +31,15 @@ inputCon <- socketConnection(
     port = port, open = "rb", blocking = TRUE, timeout = connectionTimeout)
 
 while (TRUE) {
-  ready <- socketSelect(list(inputCon))
+  ready <- socketSelect(list(inputCon), timeout = 1)
+
+  # Terminate R workers in the parent process.
+  finishedChildren <- parallel:::selectChildren()
```

Comment:
Ah, yes. It looks confusing. Probably, let me add more comments around here. I intended to do ...

A: When data is arrived in `daemon.R`:
 - `ready` is `TRUE` and terminates children, `worker.R`, if they finished their works.
 - Launch other `worker.R` children.

B: When data is not arrived in `daemon.R`:
 - `ready` is `FALSE` and terminates children, `worker.R`, if they finished their works, every second.

If we do this within `if (ready) {`, the child processes will remain until A case happens.

Forked children do some jobs and if the parent checks if any is finished right after forking by `parallel:::selectChildren()` will probably not return the finished children PIDs.